### PR TITLE
[66] As a user, I can see the list of created articles from the profile screen

### LINKED ISSUE #66: As a user, I can see the list of created articles from the profile screen
## Why
The users will be able to see the list of created articles by an author in the `Profile` screen regardless whether they are logged in or not with data provided from server.

## Acceptance Criteria
- [ ] Implement the API for getting author created articles data via this [endpoint](https://github.com/gothinkster/realworld/tree/master/api#list-articles).
- [ ] Provide `author` query parameter to filter the results accordingly the author's username.
- [ ] Setup the page size to `25`.
- [ ] Only load the first latest page and no need pagination support.
- [ ] Create a usecase for getting author created articles for later usage in integrate task.
- [ ] Parse the response into the corresponding models, here is an example response for the list of author created articles:
```
{
  "articles":[{
    "slug": "how-to-train-your-dragon",
    "title": "How to train your dragon",
    "description": "Ever wonder how?",
    "body": "It takes a Jacobian",
    "tagList": ["dragons", "training"],
    "createdAt": "2016-02-18T03:22:56.637Z",
    "updatedAt": "2016-02-18T03:48:35.824Z",
    "favorited": false,
    "favoritesCount": 0,
    "author": {
      "username": "jake",
      "bio": "I work at statefarm",
      "image": "https://i.stack.imgur.com/xHWG8.jpg",
      "following": false
    }
  }, {
    "slug": "how-to-train-your-dragon-2",
    "title": "How to train your dragon 2",
    "description": "So toothless",
    "body": "It a dragon",
    "tagList": ["dragons", "training"],
    "createdAt": "2016-02-18T03:22:56.637Z",
    "updatedAt": "2016-02-18T03:48:35.824Z",
    "favorited": false,
    "favoritesCount": 0,
    "author": {
      "username": "jake",
      "bio": "I work at statefarm",
      "image": "https://i.stack.imgur.com/xHWG8.jpg",
      "following": false
    }
  }],
  "articlesCount": 2
}
```

### PR DESCRIPTION
**Pull request by markgravity**
Resolved #66

## What happened

Add GetCreatedArticles use case

## Insight

- Update cocoapod version in `Gemfile` to match with version in `Podfile.lock`

## Proof Of Work

Request
```
*** Overview *** 
URL https://conduit.productionready.io/api/articles?author=markg&limit=25
Method GET
Response Code -
Request Start Time Sep 28 2021 - 15:14:28
Duration 848ms

*** Request Header *** 
User-Agent Nimble Medium - Staging/1.2.0 (co.nimblehq.nimble-medium-stag; build:1; iOS 14.5.0) Alamofire/5.4.4
Accept-Language en;q=1.0
Accept-Encoding br;q=1.0, gzip;q=0.9, deflate;q=0.8


*** Request Body *** 
-

*** Response Header *** 
-

*** Response Body *** 
-

------------------------------------------------------------------------
------------------------------------------------------------------------
------------------------------------------------------------------------
```

Subscribe Log
```
2021-09-28 15:14:28.482: GetCreatedArticlesUseCase -> subscribed
2021-09-28 15:14:30.611: GetCreatedArticlesUseCase -> Event next([NimbleMedium.DecodableArticle(slug: "Welcome-to-RealWorld-project", title: "Welcome to RealWorld project", description: "Exemplary fullstack Medium.com clone powered by React, Angular, Node, Django, and many more", body: "See how the exact same Medium.com clone (called Conduit) is built using different frontends and backends. Yes, you can mix and match them, because they all adhere to the same API spec", tagList: ["introduction", "welcome"], createdAt: 2021-09-25 18:17:54 +0000, updatedAt: 2021-09-25 18:17:54 +0000, favorited: false, favoritesCount: 35, apiAuthor: NimbleMedium.DecodableProfile(username: "Gerome", bio: nil, image: Optional("https://realworld-temp-api.herokuapp.com/images/demo-avatar.png"), _following: DefaultCodable.Default<DefaultCodable.False>(wrappedValue: false)))])
2021-09-28 15:14:30.611: GetCreatedArticlesUseCase -> Event completed
2021-09-28 15:14:30.611: GetCreatedArticlesUseCase -> isDisposed
```

### CODE REVIEW

**Review by minhnimble — APPROVED** [review]
LGTM 🚀

**Comment by minhnimble** [comment]
Just curious, we don't have any sample request/response log? how can you verify if the API works then 😅

File: NimbleMediumTests/Sources/Specs/Domain/UseCases/Article/GetCreatedArticlesUseCaseSpec.swift [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+import RxNimble
+import RxSwift
+import RxTest
+
+@testable import NimbleMedium
+
+final class GetCreatedArticlesUseCaseSpec: QuickSpec {
+
+    override func spec() {
+        var usecase: GetCreatedArticlesUseCase!
```
Comment (suho) [inline NimbleMediumTests/Sources/Specs/Domain/UseCases/Article/GetCreatedArticlesUseCaseSpec.swift:19]:
```suggestion
        var useCase: GetCreatedArticlesUseCase!
```

**Review by suho — APPROVED** [review]

**Review by minhnimble — CHANGES_REQUESTED** [review]
Need to fix this PR's conflict 🙏

**Review by minhnimble — APPROVED** [review]

File: NimbleMediumTests/Sources/Specs/Domain/UseCases/Article/GetCreatedArticlesUseCaseSpec.swift (hunk 1 above)
Comment (minhnimble) [reply NimbleMediumTests/Sources/Specs/Domain/UseCases/Article/GetCreatedArticlesUseCaseSpec.swift:19]: @markgravity we still have unresolved conversation here =))

**Pull request merged**